src/Behavioral.Automation/Services/Mapping/ScopeContextManager.cs

```diff
@@ -41,5 +43,14 @@ public void SwitchPage(string pageName)
             var pageScopeContext = _uriToPageScopeMapper.GetPageScopeContext(pageName);
```

Comment:
Probably, call of SwitchPage method is not needed anymore from NavigationBindings.

Reply:
Not sure, I think we still need that, in case if step is complex and we call driverService.Navigate() multiple times inside of it, Of course it may lead to extra context switches, but Andrey agreed that they are not that time consuming.